src/dune_rules/pkg_rules.ml

```diff
@@ -1972,7 +1972,14 @@ let setup_pkg_install_alias =
       | true ->
         Lock_dir.lock_dir_active ctx_name
         >>= (function
-         | false -> Memo.return Rules.empty
+         | false ->
+           let error =
+             [ Pp.text "The @pkg-install alias can't be used without a lock dir" ]
+           in
+           let hints =
+             [ Pp.text "You might want to create the lock dir with dune pkg lock" ]
+           in
+           User_error.raise ~hints error
```

Comment:
@rgrinberg would you have a suggestion on how it would be possible to raise an error when the alias is called?

If I raised an error in a `Memo.t` or in a `Rules.collect_unit`, it would simply fail when doing the check or collecting the rules. The solution I see is to make an alias which would trigger an action that would fail. I'm wondering: is there a better way to do this?

Reply:
Another alternative is to just return an action that does nothing, like an `Progn []`. Though that might make users assume `@pkg-install` did a thing when in fact it didn't do anything.

Reply:
You can use `Action_builder.fail` to delay your error